mmv1/third_party/terraform/services/storage/resource_storage_bucket.go.erb

```diff
@@ -594,7 +601,11 @@ func resourceStorageBucketCreate(d *schema.ResourceData, meta interface{}) error
 
 	err = transport_tpg.Retry(transport_tpg.RetryOptions{
 		RetryFunc: func() error {
-			res, err = config.NewStorageClient(userAgent).Buckets.Insert(project, sb).Do()
+			insertCall := config.NewStorageClient(userAgent).Buckets.Insert(project, sb)
+			if d.Get("per_object_retention").(bool) {
```

Comment:
A number of tests are failing due to this line (which ends up being 606 in the generated provider code):

```
panic: interface conversion: interface {} is nil, not bool

goroutine 9690 [running]:
github.com/hashicorp/terraform-provider-google-beta/google-beta/services/storage.resourceStorageBucketCreate.func1()
	/go/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/storage/resource_storage_bucket.go:606 +0x2df
github.com/hashicorp/terraform-provider-google-beta/google-beta/transport.Retry.func2()
	/go/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/transport/retry_utils.go:58 +0x29
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1()
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/resource/wait.go:27 +0x56
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1()
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/resource/state.go:110 +0x1ff
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/resource/state.go:83 +0x1d8
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta/services/storagetransfer	18.766s
```

I think the issue is just that this needs to be updated to match the new field name:

```suggestion
			if d.Get("enable_object_retention").(bool) {
```

Might be good to check other references as well (and for retain_until_time)

Reply:
Sorry I accepted changes and let the code stay in a bad state. The latest should build/pass just fine.